src/ol/source/WMTS.js

```diff
@@ -497,9 +497,21 @@ export function optionsFromCapabilities(wmtsCap, config) {
     const wgs84BoundingBox = l['WGS84BoundingBox'];
     const wgs84ProjectionExtent = getProjection('EPSG:4326').getExtent();
     extent = matrixSetExtent;
-    wrapX =
-      wgs84BoundingBox[0] === wgs84ProjectionExtent[0] &&
-      wgs84BoundingBox[2] === wgs84ProjectionExtent[2];
+    if (wgs84BoundingBox) {
+      wrapX =
+        wgs84BoundingBox[0] === wgs84ProjectionExtent[0] &&
+        wgs84BoundingBox[2] === wgs84ProjectionExtent[2];
+    } else {
+      const wgs84MatrixSetExtent = transformExtent(
+        matrixSetExtent,
+        'EPSG:3857',
```

Comment:
You don't know that the `matrixSetExtent` is in `EPSG:3857`. You'd need to get its CRS first.